recipes/flatcc/all/conanfile.py

```diff
@@ -0,0 +1,117 @@
+        if self.options.shared:
+            del self.options.fPIC
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        extracted_dir = self.name + "-" + self.version
+        os.rename(extracted_dir, self._source_subfolder)
+
+    def _configure_cmake(self):
+        if not self._cmake:
+            self._cmake = CMake(self)
+            self._cmake.definitions["FLATCC_PORTABLE"] = self.options.portable
+            self._cmake.definitions["FLATCC_GNU_POSIX_MEMALIGN"] = self.options.gnu_posix_memalign
+            self._cmake.definitions["FLATCC_RTONLY"] = self.options.runtime_lib_only
+            self._cmake.definitions["FLATCC_INSTALL"] = True
+            self._cmake.definitions["FLATCC_COVERAGE"] = False
+            self._cmake.definitions["FLATCC_DEBUG_VERIFY"] = self.options.verify_assert
+            self._cmake.definitions["FLATCC_TRACE_VERIFY"] = self.options.verify_trace
+            self._cmake.definitions["FLATCC_REFLECTION"] = self.options.reflection
+            self._cmake.definitions["FLATCC_NATIVE_OPTIM"] = self.options.native_optim
+            self._cmake.definitions["FLATCC_FAST_DOUBLE"] = self.options.fast_double
+            self._cmake.definitions["FLATCC_IGNORE_CONST_COND"] = self.options.ignore_const_condition
+            self._cmake.definitions["FLATCC_TEST"] = False
+            self._cmake.configure(build_folder=self._build_subfolder)
+        return self._cmake
+
+    def build(self):
+        cmake = self._configure_cmake()
+        cmake.build()
+
+
+    def package(self):
+        cmake = self._configure_cmake()
+        cmake.install()
+        if self.settings.build_type == "Debug" and not tools.os_info.is_windows:
+            debug_suffix = "_d" if self.settings.build_type == "Debug" else ""
+            exe_suffix = ".exe" if self.settings.os == "Windows" else ""
+            os.rename(os.path.join(self.package_folder, "bin", "flatcc%s%s" % (debug_suffix, exe_suffix)),
+                      os.path.join(self.package_folder, "bin", "flatcc%s" % exe_suffix))
+        # Copy license file
+        self.copy("LICENSE", dst="licenses", src=self._source_subfolder)
+
+    def package_info(self):
+        bin_path = os.path.join(self.package_folder, "bin")
+        self.output.info('Appending PATH environment variable: %s' % bin_path)
+        self.env_info.PATH.append(bin_path)
+        debug_suffix = "_d" if self.settings.build_type == "Debug" else ""
+        if not self.options.runtime_lib_only:
+            self.cpp_info.libs.append("flatcc%s" % debug_suffix)
+        self.cpp_info.libs.append("flatccrt%s" % debug_suffix)
+        if tools.os_info.is_linux:
+            self.env_info.LD_LIBRARY_PATH.append(os.path.join(self.package_folder, "lib"))
+        if tools.os_info.is_macos:
+            self.env_info.DYLD_LIBRARY_PATH.append(os.path.join(self.package_folder, "lib"))
```

Comment:
I really think this is not needed. Can you provide a scenario where you need the recipe to populate these variables?

```suggestion
```

Reply:
These lines can be removed. But my concern is that someone creates a Conan recipe for this project that uses flatcc (via cmake custom command) following the [documentation](https://docs.conan.io/en/latest/mastering/conanfile_py.html#conan-build) to create the build() function. Then when he tries to create the package he will get the dreaded `flatcc: error while loading shared libraries` might now know how to fix it.
I know the RunEnvironment is documented in the Conan reference pages but it is somewhat hidden. I actually didn't know about it until submitting this PR.

Reply:
Then we need to improve or highlight those lines in the docs (btw, using the two profiles approach, which will be the default in the future, there is no need to add the "RunEnvironment"). 

If we keep these lines here, then we need to add them to every recipe in Conan Center for the very same reason... and in that scenario, it would be something to manage inside Conan itself... and that's exactly what we have done.

Reply:
I removed the (DY)LD_LIBRARY_PATH lines. But is it still necessary to append the bin path to the PATH variable in package_info() ? Since RunEnvironment does that as well.

Reply:
@jgsogo I created a workaround in the test_package to run flatcc in the build function as you suggested, but this also fails because flatcc can't find its shared libraries. On my Hackintosh (which has SIP disabled) the test_package builds and runs fine for all build_type and 'shared' combinations.

Because dlopen searches for shared libraries in the current working directory another workaround would be to add an imports() function in the test_package recipe that copies the flatcc shared libs. But as soon as I add an imports() function the build is rejected by CI because the import() function is not allowed.
I could copy the flatcc dylib in the build function before running the flatcc executable. If you think that is acceptable than I can test that at work where we have a real Mac.